### ISSUE TITLE
Evolving Deep Neural Networks by Multi-objective Particle Swarm Optimization for Image Classification

### ISSUE THREAD

**Issue by sile** [sile/readings#76]
URL
-----

https://arxiv.org/abs/1904.09035

Abstract
----------

> In recent years, convolutional neural networks (CNNs) have become deeper in order to achieve better classification accuracy in image classification. However, it is difficult to deploy the state-of-the-art deep CNNs for industrial use due to the difficulty of manually fine-tuning the hyperparameters and the trade-off between classification accuracy and computational cost. This paper proposes a novel multi-objective optimization method for evolving state-of-the-art deep CNNs in real-life applications, which automatically evolves the non-dominant solutions at the Pareto front. Three major contributions are made: Firstly, a new encoding strategy is designed to encode one of the best state-of-the-art CNNs; With the classification accuracy and the number of floating point operations as the two objectives, a multi-objective particle swarm optimization method is developed to evolve the non-dominant solutions; Last but not least, a new infrastructure is designed to boost the experiments by concurrently running the experiments on multiple GPUs across multiple machines, and a Python library is developed and released to manage the infrastructure. The experimental results demonstrate that the non-dominant solutions found by the proposed algorithm form a clear Pareto front, and the proposed infrastructure is able to almost linearly reduce the running time. 

Table of Contents
--------------------

- (1) INTRODUCTION
  - (1-1) Goals
- (2) BACKGROUND
  - (2-1) DenseNet
  - (2-2) OMOPSO
- (3) THE PROPOSED METHOD
  - (3-1) Algorithm Overview
  - (3-2) Particle Encoding Strategy
  - (3-3) Population Initialization
  - (3-4) Objective Evaluation
  - (3-5) Infrastructure Used to Boost MOCNN
- (4) EXPERIMENT DESIGN
  - (4-1) Benchmark Dataset
  - (4-2) Parameter settings of the proposed EC methods
- (5) RESULTS AND ANALYSIS
  - (5-1) Pareto Optimality Analysis
  - (5-2) MOCNN vs DenseNet-121
  - (5-3) Computational Cost
- (6) CONCLUSIONS

BibTeX
------

```bibtex
@article{wang2019evolving,
  title={Evolving Deep Neural Networks by Multi-objective Particle Swarm Optimization for Image Classification},
  author={Wang, Bin and Sun, Yanan and Xue, Bing and Zhang, Mengjie},
  journal={arXiv preprint arXiv:1904.09035},
  year={2019}
}
```

See Also
----------

- GitHub: https://github.com/wwwbbb8510/cudam

**Comment by sile** [comment]
> However, by using the Adam optimization to train the CNNs on the whole training dataset, it could mitigate or even eliminate the bias of the aforementioned two methods because the learning rate will be automatically adapted during the training process based on the CNN architecture and the dataset, and the training process will stop until the convergence of the Adam optimization.
> (3-4. Objective Evaluation)
